### PR TITLE
Various CI fixes

### PR DESCRIPTION
**Pull request by raneamri**
Changes to merge include:
- CXX20 & CXX17 in matrix
- Small readability changes in Windows build
- Boost1.85.0 manual installation, as per QuantLib's CI, for Linux builds

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -56,8 +59,19 @@ jobs:
         path: QuantLib-Risks-Cpp
     - name: Setup
       run: |
-        sudo apt update
-        sudo apt install -y libboost-dev ccache ninja-build
+        sudo apt-get update \
```
Comment (xcelerit-team) [inline .github/workflows/ci.yaml:62]: I think we can avoid all of this if we simply use the same build container that QuantLib uses directly. Using the `container` argument - as it's done [here](https://github.com/lballabio/QuantLib/blob/master/.github/workflows/linux.yml#L136) with the `rolling` tag.  The way we follow the QuantLib pre-requisits and we don't need to install all these dependencies every time.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -315,8 +346,19 @@ jobs:
...
+        && sudo apt-get clean \
+        && sudo rm -rf /var/lib/apt/lists/*
+
+        wget https://boostorg.jfrog.io/artifactory/main/release/${{ matrix.boost_version }}/source/${{ matrix.boost_dir }}.tar.gz \
+        && tar xfz ${{ matrix.boost_dir }}.tar.gz \
+        && rm ${{ matrix.boost_dir }}.tar.gz \
+        && cd ${{ matrix.boost_dir }} \
+        && ./bootstrap.sh \
+        && sudo ./b2 --without-python --prefix=/usr -j 4 link=shared runtime-link=shared install \
+        && cd .. && sudo rm -rf ${{ matrix.boost_dir }} && sudo ldconfig
```
Comment (xcelerit-team) [inline .github/workflows/ci.yaml:361]: and the same thing here - use the container image provided by QuantLib.

**Review by xcelerit-team — CHANGES_REQUESTED** [review]

**Comment by xcelerit-team** [comment]
Can add an `apt-get install -y ninja-build` to the Linux builds? That should fix it.

**Comment by raneamri** [comment]
image

I solved the prior issue but now I'm getting a path issue. Any idea why this is happening?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -54,22 +58,23 @@ jobs:
...
-        sudo apt install -y libboost-dev ccache ninja-build
     - name: ccache
       uses: hendrikmuhs/ccache-action@v1.2.12
       with:
         key: linux-${{ matrix.disable_aad }}
         max-size: 650M
+    - name: Setup
+      run: |
+        apt-get update \
+        && apt install -y libboost-dev ccache ninja-build
```
Comment (xcelerit-team) [inline .github/workflows/ci.yaml:69]: Isn't boost already installed in the container?

**Review by xcelerit-team — CHANGES_REQUESTED** [review]
About the path issue - the problem is the `${{ github.workspace }}` doesnt resolve to the right path within the container. It point to the dir outside the container, which is not working. This is a known issue with containers - perhaps you can use something like `$(pwd)/../xad` etc to point to the right directory where xad is?

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (raneamri) [reply .github/workflows/ci.yaml:69]: It is with the rolling container, I added it back by accident. Will fix.

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -315,8 +336,19 @@ jobs:
         path: QuantLib-Risks-Cpp
     - name: Setup
       run: |
-        sudo apt update
-        sudo apt install -y libboost-dev ccache ninja-build
+        sudo apt-get update \
```
Comment (xcelerit-team) [inline .github/workflows/ci.yaml:339]: You can use the container here too

File: .github/workflows/ci.yaml [hunk 5]
```diff
@@ -39,7 +39,11 @@ jobs:
       fail-fast: false
       matrix:
         disable_aad: ["ON", "OFF"]
+        cxx: ["17", "20"]
+        boost_version: ["1.85.0"]
+        boost_dir: ["boost_1_85_0"]
```
Comment (xcelerit-team) [inline .github/workflows/ci.yaml:44]: I think these options aren't needed anymore.

File: .github/workflows/ci.yaml [hunk 6]
```diff
@@ -298,7 +314,13 @@ jobs:
         ./AdjointSwapXAD
 
   xad-linux-std-classes:
+    strategy:
+      fail-fast: false
+      matrix:
+        boost_version: ["1.85.0"]
+        boost_dir: ["boost_1_85_0"]
```
Comment (xcelerit-team) [inline .github/workflows/ci.yaml:321]: You don't need a matrix build here.

**Comment by raneamri** [comment]
This is ready to merge. The windows cxx20 failure is not part of this pull request.

**Comment by xcelerit-team** [comment]
> This is ready to merge. The windows cxx20 failure is not part of this pull request.

Then please disable that workflow (comment it or put an if false). Can't merge with failing CI/CD.

**Review by auto-differentiation-dev — APPROVED** [review]

**Review by xcelerit-team — APPROVED** [review]

**Pull request merged**